modules/dnn/test/test_onnx_importer.cpp

```diff
@@ -2669,6 +2669,89 @@ TEST_P(Test_ONNX_layers, YOLOv7)
     normAssertDetections(refClassIds, refScores, refBoxes, keep_classIds, keep_confidences, keep_boxes);
 }
 
+
+TEST_P(Test_ONNX_layers, YOLOv5m)
+{
+    std::string weightPath = _tf("models/yolov5m.onnx", false);
+    std::string imgPath = _tf("../dog_orig_size.png");
+
+    Size targetSize{640, 640};
+    float conf_threshold = 0.25;
+    float iou_threshold = 0.45;
+
+    // Reference, which is collected with input size of 640x640
+    std::vector<int> refClassIds{16, 1, 7};
+    std::vector<float> refScores{0.907127f, 0.889226f, 0.74817f};
+    // [x1, y1, x2, y2] x 4
+
+    std::vector<Rect2d> refBoxes{Rect2d(109.318f, 252.709f, 257.87f, 598.693f),
+                                 Rect2d(102.513f, 146.225f, 478.182f, 464.939f),
+                                 Rect2d(392.139f, 82.1886, 578.795f, 190.488f)};
+
+    Mat img = imread(imgPath);
+    Mat inp = blobFromImage(img, 1/255.0, targetSize, Scalar(0, 0, 0), true, false);
+
+    Net net = readNet(weightPath);
+
+    net.setInput(inp);
+    std::vector<Mat> outs;
+    net.forward(outs, net.getUnconnectedOutLayersNames());
+    Mat preds = outs[0].reshape(1, outs[0].size[1]); // [1, 25200, 85]
+
+    // Retrieve
+    std::vector<int> classIds;
+    std::vector<float> confidences;
+    std::vector<Rect2d> boxes;
+    // each row is [cx, cy, w, h, conf_obj, conf_class1, ..., conf_class80]
+    for (int i = 0; i < preds.rows; ++i)
+    {
+        // filter out non objects
+        float obj_conf = preds.row(i).at<float>(4);
+        if (obj_conf < conf_threshold)
+            continue;
+
+        // get class id and conf
+        Mat scores = preds.row(i).colRange(5, preds.cols);
+        double conf;
+        Point maxLoc;
+        minMaxLoc(scores, 0, &conf, 0, &maxLoc);
+        conf *= obj_conf;
+        if (conf < conf_threshold)
+            continue;
+
+        // get bbox coords
+        float* det = preds.ptr<float>(i);
+        double cx = det[0];
+        double cy = det[1];
+        double w = det[2];
+        double h = det[3];
+        // [x1, y1, x2, y2]
+        boxes.push_back(Rect2d(cx - 0.5 * w, cy - 0.5 * h,
+                                cx + 0.5 * w, cy + 0.5 * h));
+        classIds.push_back(maxLoc.x);
+        confidences.push_back(conf);
+    }
+
+    // NMS
+    std::vector<int> keep_idx;
+    NMSBoxes(boxes, confidences, conf_threshold, iou_threshold, keep_idx);
+
+    std::vector<int> keep_classIds;
+    std::vector<float> keep_confidences;
+    std::vector<Rect2d> keep_boxes;
+    for (auto i : keep_idx)
+    {
+        keep_classIds.push_back(classIds[i]);
+        keep_confidences.push_back(confidences[i]);
+        keep_boxes.push_back(boxes[i]);
+    }
+    for (int i = 0; i <= 2; i++) 
+        cout << keep_classIds[i];
```

Comment:
Please remove it from test.

Reply:
@dkurt As you recommended, I checked yolov5n:

1. On my test image it works worse than yolov5m: bicycle box is predicted incorrectly.
2. Opencv result detection result on my test image differs from ultralytics result: scores are lower

I attach ultralitics detection visualization and my visualization of the same model (yolov5n.onnx) for the same image.

In [ultralytis](https://github.com/ultralytics/yolov5) authors also report better performance of yolov5m in comparison with yolov5n.

Should I still add a test for yolov5n? Or should I add  append opencv_extra with yolov5m link?



![ultralytics_result](https://github.com/opencv/opencv/assets/39569266/ed4c85bc-8c71-4dea-981d-e2556a085fdd)
![opencv_result](https://github.com/opencv/opencv/assets/39569266/e6cb8576-8d23-4695-bddf-5faadb7c91c8)

Reply:
Let’s start from the motivation. If both 5n and 5m models consist of the same layers and there is no difference in the testing perspective which one to test - use 5n. Tests are needed to cover OpenCV implementation. For demonstration object_detection.py sample should be used.

So if you can, please keep test run 5n but put a link to yopo5m to the sample as it’s done here: https://github.com/opencv/opencv/pull/24496